com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/MyLanguageServerImpl.java

```diff
@@ -64,19 +64,19 @@
 
   private TextDocumentService textService;
   private CobolWorkspaceService workspaceService;
-  private FileSystemService fileSystemService;
+  private CopybookService copybookService;
   private Provider<LanguageClient> clientProvider;
-  private Provider<ConfigurationSettingsStorable> settingsProvider;
+  private Provider<SettingsProvider> settingsProvider;
 
   @Inject
   MyLanguageServerImpl(
-      FileSystemService fileSystemService,
+      CopybookService copybookService,
       TextDocumentService textService,
       CobolWorkspaceService workspaceService,
       Provider<LanguageClient> clientProvider,
-      Provider<ConfigurationSettingsStorable> settingsProvider) {
+      Provider<SettingsProvider> settingsProvider) {
```

Comment:
Why do you use a provider for provider? Injecting SettingsProvider is enough here